packages/core/src/components/button/_button-mixin.scss

```diff
@@ -0,0 +1,453 @@
+  --computed-border-color-active: var(--border-color-active, var(--computed-border-color-hover, transparent));
+  --computed-border-color-disabled: var(--border-color-disabled, var(--computed-background-color-disabled, transparent));
+  --computed-outline-color-focus: var(--outline-color-focus, transparent);
+  --computed-icon-size: var(--icon-size, var(--spacing-m));
+
+  align-content: flex-start;
+  align-items: center;
+  background-color: var(--computed-background-color);
+  border: var(--border-width) solid var(--computed-border-color);
+  border-radius: 0;
+  -moz-box-sizing: border-box;
+  -webkit-box-sizing: border-box;
+  box-sizing: border-box;
+  color: var(--computed-color, inherit);
+  column-gap: var(--spacing-2-xs);
+  cursor: var(--cursor, pointer);
+  display: inline-flex;
+  font-family: inherit;
+  font-size: 100%;
+  font-weight: 500;
+  hyphens: auto;
+  justify-content: center;
+  line-height: var(--lineheight-l);
+  margin: 0;
+  min-height: var(--min-size);
+  min-width: var(--min-size);
+  outline: none;
+  outline-offset: 2px;
+  overflow: hidden;
+  overflow-wrap: anywhere;
+  padding: var(--padding);
+  position: relative;
+  text-align: center;
+  text-decoration: none;
+  text-transform: none;
+  vertical-align: top;
+
+  /* repositioning the icons, after & before usage */
+  &:after {
+    margin-right: var(--icon-reposition);
+  }
+
+  &:before {
+    margin-left: var(--icon-reposition);
+  }
+
+  &,
+  &[type="button"],
+  &[type="reset"],
+  &[type="submit"]
+  {
+    -webkit-appearance: button;
+  }
+
+  /* focus */
+  &:focus-visible {
+    background-color: var(--computed-background-color-focus);
+    border-color: var(--computed-border-color-focus);
+    color: var(--computed-color-focus);
+    outline: var(--outline-width-focus) solid var(--computed-outline-color-focus);
+  }
+
+  /* hover */
+  &:hover {
+    background-color: var(--computed-background-color-hover);
+    border-color: var(--computed-border-color-hover);
+    color: var(--computed-color-hover);
+  }
+
+  /* active */
+  &:active {
+    background-color: var(--computed-background-color-active);
+    border-color: var(--computed-border-color-active);
+    color: var(--computed-color-active);
+    outline: var(--outline-width-focus) solid var(--computed-outline-color-focus);
+  }
+
+  /* disabled */
+  &:disabled {
+    --cursor: not-allowed;
+
+    background-color: var(--computed-background-color-disabled);
+    border-color: var(--computed-border-color-disabled);
+    color: var(--computed-color-disabled);
+    outline: none;
+  }
+}
+
+@mixin icon {
+  & .hds-icon {
+    flex-shrink: 0;
+    height: var(--computed-icon-size);
+    width: var(--computed-icon-size);
+  }
+
+  /* start-side icon */
+  & .hds-icon:first-child:not(:last-child) {
+    margin-left: var(--icon-reposition);
+  }
+
+  /* end-side icon */
+  & .hds-icon:last-child:not(:first-child) {
+    margin-right: var(--icon-reposition);
+  }
+}
+
+@mixin iconReact {
```

Comment:
Hmm, is there no way to handle this without duplication (@mixin icon & iconReact)?

Reply:
simplified